packages/block-editor/src/components/rich-text/use-should-disable-editing.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { getBlockType } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { BLOCK_BINDINGS_ALLOWED_BLOCKS } from '../../hooks/use-bindings-attributes';
+import { useBlockEditContext } from '../block-edit';
+import { store as blockEditorStore } from '../../store';
+import { unlock } from '../../lock-unlock';
+
+export function useShouldDisableEditing() {
+	const { clientId, name: blockName } = useBlockEditContext();
+
+	const { getBlockAttributes } = useSelect( blockEditorStore );
+	const { getBlockBindingsSource } = unlock( useSelect( blockEditorStore ) );
+	const blockBindings = getBlockAttributes( clientId )?.metadata?.bindings;
+
+	if ( blockBindings && blockName in BLOCK_BINDINGS_ALLOWED_BLOCKS ) {
+		const blockTypeAttributes = getBlockType( blockName ).attributes;
+
+		for ( const [ attribute, args ] of Object.entries( blockBindings ) ) {
+			if ( blockTypeAttributes?.[ attribute ]?.source !== 'rich-text' ) {
+				break;
+			}
+
+			// If the source is not defined, or if its value of `lockAttributesEditing` is `true`, disable it.
+			const blockBindingsSource = getBlockBindingsSource( args.source );
+			if (
+				! blockBindingsSource ||
+				blockBindingsSource.lockAttributesEditing
+			) {
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
```

Comment:
I was playing with the potential refactoring here where the `identifier` prop would get passed as an attribute name:
```suggestion
export function useShouldDisableEditing( attributeName ) {
	const { clientId, name: blockName } = useBlockEditContext();

	return useSelect(
		( select ) => {
			if (
				! attributeName ||
				! BLOCK_BINDINGS_ALLOWED_BLOCKS[ blockName ]?.includes(
					attributeName
				)
			) {
				return false;
			}

			const blockBindings =
				select( blockEditorStore ).getBlockAttributes( clientId )
					?.metadata?.bindings;
			if ( ! blockBindings?.[ attributeName ]?.source ) {
				return false;
			}

			const blockBindingsSource = unlock(
				select( blockEditorStore )
			).getBlockBindingsSource( blockBindings[ attributeName ].source );
			return blockBindingsSource?.lockAttributesEditing !== false;
		},
		[ clientId, blockName, attributeName ]
	);
}
```

Reply:
Added with https://github.com/WordPress/gutenberg/pull/58958/commits/5578d8c20fcdcd924001197a6b2facbf25b9f96f. If something goes wrong, we can always remove the commit. In my testing, it's all good.